### ISSUE TITLE
`avoid_renaming_method_parameters` does not cover a class which implements another.

### ISSUE THREAD

**Issue by srawlins** [dart-lang/linter#4038]
**Describe the issue**

Element documentation can come from a superclass which is extended, or which is implemented (or, presumably, which is mixed in). But this lint rule does not catch this case. See on the [page for List](https://api.dart.dev/stable/2.19.1/dart-core/List-class.html) that docs are inherited from Iterable, for `any`, for example. List implements Iterable.

**To Reproduce**

```dart
abstract class Foo {
  /// Tell me about [x].
  void m(int x);
}

class Bar implements Foo {
  @override
  void m(int y) {}
}
```

**Expected behavior**

Lint.
